src/passwd.c

```diff
@@ -506,11 +501,14 @@ oom (void)
 
 static char *update_crypt_pw (char *cp)
 {
-#ifndef USE_PAM
-	if (do_update_pwd) {
-		cp = xstrdup (crypt_passwd);
+#ifdef USE_PAM
+	if (!use_pam)
+#endif				/* USE_PAM */
```

Comment:
If you already set `use_pam` when `USE_PAM` is defined, then why do you need to build this conditionally?

Reply:
I think https://github.com/shadow-maint/shadow/pull/714#issuecomment-1554849599 and https://github.com/shadow-maint/shadow/pull/714#issuecomment-1555167191 are about this as well.

Reply:
Fixed. I already had the changes stages since when Hallyn mentioned this, the original reason was that I tried to be consistent with chpasswd (which does this).

Reply:
I prefer to have a neater code than consistency with other tools.